File: 프로그래머스/Level 2/124 나라의 숫자.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12899
def solution(n):
    answer = ''

    # 3진법 이용
    # 0 대신 4로
    while n:
        if n % 3:
            answer += str(n % 3)
            n //= 3
        else:
            answer += "4"
            n //= 3
            n -= 1  # 이를 안해주면 1 들어가짐
    return answer[::-1]

File: 프로그래머스/Level 2/124 나라의 숫자.swift
// https://school.programmers.co.kr/learn/courses/30/lessons/12899

func solution(_ n: Int) -> String {
    var answer:String = ""
    var tmpArr:[String] = []
    var number = n

    while n > 0 {
        if n % 3 == 1 {
            tmpArr.append(String(n%3))
            n /= 3
        } else {
            tmpArr.append("4")
            n /= 3
            n -= 1
        }
    }

    answer = tmpArr.reversed().joined()
    return answer
}